analyzer/codechecker_analyzer/analyzers/clangsa/analyzer.py

```diff
@@ -379,4 +395,29 @@ def construct_config_handler(cls, args, context):
             cmdline_checkers,
             'enable_all' in args and args.enable_all)
 
+        handler.checker_config = []
+        r = re.compile(r'(?P<analyzer>.+?):(?P<key>.+?)=(?P<value>.+)')
+
+        # TODO: This extra "isinstance" check is needed for
+        # CodeChecker checkers --checker-config. This command also runs
+        # this function in order to construct a config handler.
+        if 'checker_config' in args and \
```

Comment:
If this option is given in the command line it will be always a list. If you remove the `default=argparse.SUPPRESS` from the argparse you can change this condition to: `if args.checker_config:`.

Reply:
Yes, this was my original solution. However, unfortunately this `construct_config_handler()` function is called when `CodeChecker checkers --checker-config` command is run and in that sub-command this flag has no argument. In this case the value of `args.checker_config` is `True`. After fixing the TODO above this can also be fixed.